LGPE support

6A-Realm opened the pull request:
Added LGPE support with additional minimal changes.

Only showdown implementation is tested and verified. Please note all changed, I'm assuming not all is wanted.

6A-Realm committed:
LGPE support

Added LGPE support with additional minimal changes.

DevYukine commented on Pokégen.PkHex/Controllers/PokemonController.cs:
```diff
@@ -125,7 +125,7 @@ public async Task<IActionResult> CheckUrlLegality([FromRoute] string game, [From
 	[Route("file")]
 	[ProducesResponseType(StatusCodes.Status200OK)]
 	[ProducesResponseType(StatusCodes.Status400BadRequest)]
-	[RequestFormLimits(MultipartBodyLengthLimit = 344)]
```
this change is wrong as its a **Limit** and not a minimum, please revert this

DevYukine commented on Pokégen.PkHex/Controllers/PokemonController.cs:
```diff
@@ -145,7 +145,7 @@ public async Task<IActionResult> GetPokemonFromFile([FromRoute] string game, [Fr
 	[Route("file/legality")]
 	[ProducesResponseType(StatusCodes.Status204NoContent)]
 	[ProducesResponseType(StatusCodes.Status400BadRequest)]
-	[RequestFormLimits(MultipartBodyLengthLimit = 344)]
```
Same as other comment on same file

DevYukine commented on Pokégen.PkHex/Program.cs:
```diff
@@ -27,7 +27,7 @@
 		.Enrich.FromLogContext()
 		.WriteTo.Console(
 			outputTemplate:
-			"[{Timestamp:yyyy-MM-dd HH:mm:ss} {Level:u3}] [{ThreadName}] {Message:lj}\n{Exception}",
```
Please do not change the logging output which is also not part of this PR

DevYukine commented on Pokégen.PkHex/Program.cs:
```diff
@@ -66,11 +66,11 @@
 
 builder.Services.AddSwaggerGen(c =>
 {
-	c.SwaggerDoc("v1", new OpenApiInfo
```
Also revert this as we aren't on v2 yet

DevYukine commented on Pokégen.PkHex/Properties/launchSettings.json:
```diff
@@ -25,9 +25,9 @@
       "applicationUrl": "https://localhost:5001;http://localhost:5000",
       "environmentVariables": {
 		  "ASPNETCORE_ENVIRONMENT": "Development",
-		  "PKHEX_DEFAULT_OT": "PkHex",
```
Do not change defaults in here

DevYukine commented on Pokégen.PkHex/Services/DownloaderService.cs:
```diff
@@ -18,8 +18,8 @@ public DownloaderService(HttpClient httpClient)
 		
 	public async Task<PKM> DownloadPkmAsync(Uri uri, long? length, SupportedGame wantedGame)
 	{
-		long? size = null;
```
do not rename variables names in here which do not match the project style settings

DevYukine commented on Pokégen.PkHex/Services/PokemonService.cs:
```diff
@@ -40,15 +42,11 @@ public async Task<PKM> GetPokemonFromFormFileAsync(IFormFile file, SupportedGame
 	public Task<PKM> GetPokemonFromShowdown(string showdownSet, SupportedGame game)
 	{
 		var set = new ShowdownSet(showdownSet);
-		var template = set.GetTemplate();
```
Do not remove the error handling in here as its needed for ASP.Net to properly recognize this an a non 500 exception

DevYukine commented on Pokégen.PkHex/Services/PokemonService.cs:
```diff
@@ -40,15 +42,11 @@ public async Task<PKM> GetPokemonFromFormFileAsync(IFormFile file, SupportedGame
 
 		var sav = game switch
 		{
+			SupportedGame.LGPE => SaveUtil.GetBlankSAV(GameVersion.GE, "SysBot.py"),
```
Please do the same as i did for the others with `AutoLegalityModService.GetTrainerInfo` for the specific type

DevYukine reviewed (changes requested):
Overall please revert the Linebreak changes and stick to the style settings in the .editorconfig (you might need to download a plugin for your IDE to recognize this file)

also please do not change names of PkHex -> PKHeX, while its a correct change its out of scope for this PR and you should do it in an own PR, so please revert that as well + all my requested changes

6A-Realm committed:
Delete .github/workflows directory

6A-Realm committed:
Deleted Pokegen directory

Preparing for new directory.

6A-Realm committed:
Delete .editorconfig

6A-Realm committed:
Delete .gitignore

6A-Realm committed:
Delete LICENSE

6A-Realm committed:
Delete Pokégen.PkHex.sln

6A-Realm committed:
Delete Pokégen.PkHex.sln.DotSettings

6A-Realm committed:
Full PKHeX-API support

- Tested support for LGPE, SWSH, BDSP, PLA.
- PKHeX-API rebranding.
- Removal of all unused lines of code.
- Public deployment methods.

6A-Realm committed:
ALM update

Update dependencies Auto-Legality Mod 22.3.18

6A-Realm committed:
Update PKHeX-API.csproj
